### ISSUE TITLE
Position of `all` property is not maintained

### ISSUE THREAD

**Issue by JimDabell** [parcel-bundler/lightningcss#746]
I have the following CSS:

```css
button {
  all: unset;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  gap: 0.375em;
  margin: 0;
  padding: 0.125em 0.5em;
  background: var(--post-background-lighter);
  border: thin solid var(--post-background-darker);
  border-radius: 1em;
  cursor: pointer;
}
```

When rendered with v1.24.1, the position of the `all` property is maintained:

```css
button[data-v-77fddb74] {
  all: unset;
  box-sizing: border-box;
  background: var(--post-background-lighter);
  border: thin solid var(--post-background-darker);
  cursor: pointer;
  border-radius: 1em;
  align-items: center;
  gap: .375em;
  margin: 0;
  padding: .125em .5em;
  display: flex;
}
```

When rendered with v1.25.0, the position of the `all` property changes, breaking styles:

```css
button[data-v-77fddb74] {
  box-sizing: border-box;
  background: var(--post-background-lighter);
  border: thin solid var(--post-background-darker);
  cursor: pointer;
  all: unset;
  border-radius: 1em;
  align-items: center;
  gap: .375em;
  margin: 0;
  padding: .125em .5em;
  display: flex;
}
```

I see a relevant commit [d7aeff3](https://github.com/parcel-bundler/lightningcss/commit/d7aeff3db67ee9d15e0fefce9251cf41c0b8ec44) _optimize "all" property_, that has a comment:

> // Always place the `all` property first. Previous properties will have been omitted.

This doesn’t appear to be correct.

**Comment by devongovett** [comment]
Yeah this is a bad bug, sorry about that. I missed part of this. Will push a patch out asap

**Comment by devongovett** [comment]
Fixed in v1.25.1. My apologies for the bug!

**Comment by JimDabell** [comment]
I can confirm that’s fixed for me in v1.25.1. Thanks for the quick turnaround!
